### PR TITLE
Stop putting suite loads into active test

### PR DESCRIPTION
**Pull request by natebosch**
Towards #1311

Removes a UX concern from `Engine` which no longer needs to worry about
how a reporter might display information about load suites. Makes the
interaction between the engine and reporters less subtle.

Use the term "suite load" over "load test" to start pushing terminology
away from conflating tests and loading.

### CODE REVIEW

File: pkgs/test_core/lib/src/runner/engine.dart [hunk 1]
```diff
@@ -174,22 +169,22 @@ class Engine {
   Set<LiveTest> get failed => _failedGroup.set;
   final _failedGroup = UnionSetController<LiveTest>(disjoint: true);
 
-  /// The tests that are still running, in the order they begain running.
+  /// The tests that are still running, in the order they began running.
   List<LiveTest> get active => UnmodifiableListView(_active);
   final _active = QueueList<LiveTest>();
 
+  /// The tests from [LoadSuite]s that are still running, in the order they
+  /// began running.
```
Comment (jakemac53) [inline pkgs/test_core/lib/src/runner/engine.dart:177]: Might as well change the wording here to stop referring to them as "tests" as well.

```suggestion
  /// The [LoadSuite]s that are still loading, in the order they began.
```

File: pkgs/test_core/lib/src/runner/reporter/compact.dart [hunk 2]
```diff
@@ -186,10 +186,12 @@ class CompactReporter implements Reporter {
...
-    // If this is the first test to start, print a progress line so the user
-    // knows what's running. It's possible that the active test may not be
-    // [liveTest] because the engine doesn't always surface load tests.
-    if (_engine.active.length == 1 && _engine.active.first == liveTest) {
+    // If this is the first test or suite load to start, print a progress line
+    // so the user knows what's running.
+    if ((_engine.active.length == 1 && _engine.active.first == liveTest) ||
+        (_engine.liveTests.isEmpty &&
+            _engine.activeSuiteLoads.length == 1 &&
+            _engine.activeSuiteLoads.first == liveTest)) {
```
Comment (jakemac53) [inline pkgs/test_core/lib/src/runner/reporter/compact.dart:194]: I don't think  you want the `liveTests` check here - that would make it not show the loading message for the currently loading test if some test has completed and some are still loading, but none is actively running.

```suggestion
        (_engine.activeSuiteLoads.length == 1 &&
            _engine.activeSuiteLoads.first == liveTest)) {
```

Comment (natebosch) [reply pkgs/test_core/lib/src/runner/reporter/compact.dart:194]: Ah, yeah this should have been `_engine.active.isEmpty`, not `_engine.liveTests.isEmpty`.

File: pkgs/test_core/lib/src/runner/engine.dart (hunk 1 above)
Comment (natebosch) [reply pkgs/test_core/lib/src/runner/engine.dart]: Done.

File: pkgs/test_core/lib/src/runner/reporter/compact.dart (hunk 2 above)
Comment (natebosch) [reply pkgs/test_core/lib/src/runner/reporter/compact.dart]: We don't want to remove the check entirely or we'd show info about suites that would have never been put into `_enginge.active` before.

**Review by jakemac53 — APPROVED** [review]

**Pull request merged**